File: src/Aiursoft.Kahla.SDK/Events/ThreadPropertyChangedEvent.cs
using Aiursoft.Kahla.SDK.Events.Abstractions;

namespace Aiursoft.Kahla.SDK.Events;

public class ThreadPropertyChangedEvent : KahlaEvent
{
    public ThreadPropertyChangedEvent()
    {
        Type = EventType.ThreadPropertyChanged;
    }

    public required int ThreadId { get; init; }
    
    public required string ThreadName { get; init; }
    
    public required string ThreadImagePath { get; init; }
}

File: src/Aiursoft.Kahla.Frontend/kahla.app/src/app/Models/Events/ThreadPropertyChangedEvent.ts
import { KahlaEvent } from './KahlaEvent';
import { KahlaEventType } from './EventType';

export interface ThreadPropertyChangedEvent extends KahlaEvent {
    type: KahlaEventType.ThreadPropertyChanged;
    threadId: number;
    threadName: string;
    threadImagePath: string;
}

export function isThreadPropertyChangedEvent(event: KahlaEvent): event is ThreadPropertyChangedEvent {
    return event.type === KahlaEventType.ThreadPropertyChanged;
}
